frame/society/src/lib.rs

```diff
@@ -0,0 +1,1372 @@
+		/// The minimum amount of a deposit required for a bid to be made.
+		const CandidateDeposit: BalanceOf<T, I> = T::CandidateDeposit::get();
+
+		/// The amount of the unpaid reward that gets deducted in the case that either a skeptic
+		/// doesn't vote or someone votes in the wrong way.
+		const WrongSideDeduction: BalanceOf<T, I> = T::WrongSideDeduction::get();
+
+		/// The number of times a member may vote the wrong way (or not at all, when they are a skeptic)
+		/// before they become suspended.
+		const MaxStrikes: u32 = T::MaxStrikes::get();
+
+		/// The amount of incentive paid within each period. Doesn't include VoterTip.
+		const PeriodSpend: BalanceOf<T, I> = T::PeriodSpend::get();
+
+		/// The number of blocks between candidate/membership rotation periods.
+		const RotationPeriod: T::BlockNumber = T::RotationPeriod::get();
+
+		/// The number of blocks between membership challenges.
+		const ChallengePeriod: T::BlockNumber = T::ChallengePeriod::get();
+
+		// Used for handling module events.
+		fn deposit_event() = default;
+
+		/// A user outside of the society can make a bid for entry.
+		///
+		/// Payment: `CandidateDeposit` will be reserved for making a bid. It is returned
+		/// when the bid becomes a member, or if the bid calls `unbid`.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// Parameters:
+		///
+		/// - `value`: A one time payment the bid would like to receive when joining the society.
+		///
+		/// # <weight>
+		/// - Storage Reads:
+		/// 	- One storage read to check for suspended candidate.
+		/// 	- One storage read to check for suspended member.
+		/// 	- One storage read to retrieve all current bids.
+		/// 	- One storage read to retrieve all current candidates.
+		/// 	- One storage read to retrieve all members.
+		/// - O(B + C + log M) to check user is not already a part of society.
+		/// - External Module Operations:
+		/// 	- One balance reserve operation.
+		/// - Write Operations:
+		/// 	- 
+		/// - One event.
+		/// # </weight>
+
+		/// Alters Bids (O(N) decode+encode, O(logN) search, <=2*O(N) write).
+		pub fn bid(origin, value: BalanceOf<T, I>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(!<SuspendedCandidates<T, I>>::exists(&who), Error::<T, I>::Suspended);
+			ensure!(!<SuspendedMembers<T, I>>::exists(&who), Error::<T, I>::Suspended);
+			ensure!(!Self::is_bid(&who), Error::<T, I>::AlreadyBid);
+			ensure!(!Self::is_candidate(&who), Error::<T, I>::AlreadyCandidate);
+			ensure!(!Self::is_member(&who), Error::<T, I>::AlreadyMember);
+
+			let deposit = T::CandidateDeposit::get();
+			T::Currency::reserve(&who, deposit)?;
+
+			Self::put_bid(who.clone(), value.clone(), BidKind::Deposit(deposit));
+			Self::deposit_event(RawEvent::Bid(who, value));
+			Ok(())
+		}
+
+		/// A bid can remove their bid for entry into society.
+		///
+		/// By doing so, they will have their candidate deposit returned or
+		/// they will unvouch their voucher.
+		///
```

Comment:
Should mention the payment it will take here with a `Payment: ` prefixed line. see `identity` module for examples.